Updating API Documentation for Tasks and Tags

madpink opened the pull request:
[//]: # (Note: See http://habitica.wikia.com/wiki/Using_Habitica_Git#Pull_Request for more info)

https://github.com/HabitRPG/habitica/issues/8087

### Changes
Beginning the processes of updating Tasks
Added Body parameters
Added Request example
Added Success example
Added Failure example
Expanded error types


[//]: # (Put User ID in here - found in Settings -> API)

----
UUID:

myoshuGO committed:
Fixes #8227 (#8380)

* Fixes #8227

* Fixes #8227

* Fixes #8227

ramblinjan committed:
Fix home column style on medium screen widths fixes #8372 (#8387)

* handle medium screen width on home columns

* rename new-row helper used on home screen for todo column flow

declanramsay committed:
Fixes task left in odd setting when cancelling editing with ESC / Escape Key; add click outside to close modal #8308 #8321 (#8382)

* Remove backdrop property from open modal call to allow click-outside to close, call cancelTaskEdit function on dismissal of edit modal via ESC or click-outside

* Remove commented out code, change catch function to normal (non-arrow)

* Modify task services test mock openModal function, to handle new use of subsequent promise resolution functionality

* Tidy old edit controls away

SabreCat committed:
feat(content): Triceratops pet quest

SabreCat committed:
chore(migration): update Bailey to use monk

SabreCat committed:
3.67.0

SabreCat committed:
fix(coveralls): update badge for /habitica

Mathwhiz committed:
News translation link (#8393)

* Add translation link to news

* Add newsArchive string

* Translate news archive link

* Fix link?

* Add link to wiki

TheHollidayInn committed:
Group plans reorder tasks (#8358)

* Added move route for group tasks

* Added group task reorder to front end

* Added syncing with group task order

* Fixed linting issues

* Added missing exec and abstracted move code

* Added unit test for moveTask

madpink committed:
Update tasks.js

TheHollidayInn commented on website/server/controllers/api-v3/tasks.js:
```diff
@@ -41,7 +41,103 @@ let requiredGroupFields = '_id leader tasksOrder name';
+ * @apiParam (Body) {string} [notes] Extra notes 
+ * @apiParam (Body) {number} [priority=1] Difficulty, options are 0.1, 1, 1.5, 2; eqivalent of Trivial, Easy, Medium, Hard.
+ * @apiParam (Body) {string[]} [reminders] Array of reminder data including a UUID, startDate and time
+ * @apiParam (Body) {string} [reminders.id] UUID of the reminder
```
This seems a little odd. You won't have an array of just a reminder.id. I think it would be better to show `[{id: 'example'}]` or something to that effect.

madpink replied:
Would it make more sense to describe reminders as below?
* @apiParam (Body) {string[]} [reminders] Array of reminders, each being an object containing id, startDate and time

Mathwhiz committed:
Update Royal Purple Flying PIg (#8401)

SabreCat committed:
fix(incentives): return 0 for upcoming at end of list
Also adds some new strings for upcoming content, adds a news announcement, and compiles a recent sprites fix.

SabreCat committed:
chore(i18n): update locales

SabreCat committed:
3.67.1

madpink committed:
Update API Doc for CreateUserTasks

madpink commented:
Changes:
-removed "reminders" array keys (id, startDate, time) and included them in the "reminders" description
-added allowed values to types where applicable
-added date body param

Forgot to put this in:
  * UUID: b0413351-405f-416f-8787-947ec1c85199

paglias committed:
chore(i18n): update locales

paglias committed:
chore(locales): add merch file

madpink committed:
Added value to body params for "reward" type

TheHollidayInn commented:
Looks good to me!

Mathwhiz committed:
apiDoc: meta (#8167)

* apiDoc: meta

* Update modelsPaths.js

* Update error

* fix test

* Update GET-model_paths.test.js

* Fixed test

madpink committed:
Update tasks.js

madpink commented:
About half done updating

paglias committed:
New Client: english translations and misc fixes (#8410)

* misc fixes and add english translations

* add tests

khipkin commented on website/server/controllers/api-v3/tasks.js:
```diff
@@ -62,17 +163,47 @@ api.createUserTasks = {
 };
 
-/**
+/**   QZQZ TODO
```
Don't forget to remove this! 😛

paglias committed:
client: remove old code

madpink committed:
Updated API Documentation for tasks.js

Finished going through all of the API calls and updating documentation.

madpink commented:
API Documentation update complete

madpink commented:
I'm looking over tasks.js and I don't see the score commands... am I missing a chunk of the file?

madpink committed:
Updated tasks.js - added score example

khipkin commented on website/server/controllers/api-v3/tasks.js:
```diff
@@ -481,13 +481,19 @@ api.updateTask = {
  * {"success":true,"data":{"delta":0.9746999906450404,"_tmp":{},"hp":49.06645205596985,"mp":37.2008917491047,"exp":101.93810026267543,"gp":77.09694176716997,"lvl":19,"class":"rogue","points":0,"str":5,"con":3,"int":3,"per":8,"buffs":{"str":9,"int":9,"per":9,"con":9,"stealth":0,"streaks":false,"snowball":false,"spookySparkles":false,"shinySeed":false,"seafoam":false},"training":{"int":0,"per":0,"s ...
  *
+ * @apiSuccessExample {json} Example result with item drop:
+ * {"success":true,"data":{"delta":1.0259567046270648,"_tmp":{"quest":{"progressDelta":1.2362778290756147,"collection":1},"drop":{"target":"Zombie","article":"","canDrop":true,"value":1,"key":"RottenMeat","type":"Food","dialog":"You've found Rotten Meat! Feed this to a pet and it may grow into a sturdy steed."}},"hp":50,"mp":66.2390716654227,"exp":143.93810026267545,"gp":135.12889840462591,"lvl": ...
```
This is a super duper long line. @staff Is that okay?

madpink committed:
Updated tags.js API documentation

Unable to update "reorder-tags", unable to successfully use

madpink committed:
Updated Tags APIDoc

Unable to get a working example of reorder-tags

Alys committed:
remove outdated information about changing your email address

This is being done to remove a redundant string that
contains an email address, in preparation for actioning
https://github.com/HabitRPG/habitica/issues/8385 ("Email addresses
should not probably be translatable")

paglias committed:
client: add missing test files and use v-once

paglias committed:
client: replace deprecated vue-resource with axios, lint more file

librarianmage reviewed (changes requested):
Make sure to check the [tests](https://travis-ci.org/HabitRPG/habitica/jobs/192017738) for your latest commit. It appears you have some trailing spaces in your code, which causes the `lint` test to fail:

```
/home/travis/build/HabitRPG/habitica/website/server/controllers/api-v3/tasks.js
   56:49  error  Trailing spaces not allowed  no-trailing-spaces
   57:99  error  Trailing spaces not allowed  no-trailing-spaces
   72:61  error  Trailing spaces not allowed  no-trailing-spaces
   98:3   error  Trailing spaces not allowed  no-trailing-spaces
  102:3   error  Trailing spaces not allowed  no-trailing-spaces
  111:3   error  Trailing spaces not allowed  no-trailing-spaces
  117:3   error  Trailing spaces not allowed  no-trailing-spaces
  124:3   error  Trailing spaces not allowed  no-trailing-spaces
  167:4   error  Trailing spaces not allowed  no-trailing-spaces
  180:49  error  Trailing spaces not allowed  no-trailing-spaces
  181:99  error  Trailing spaces not allowed  no-trailing-spaces
  373:49  error  Trailing spaces not allowed  no-trailing-spaces
  374:99  error  Trailing spaces not allowed  no-trailing-spaces
  618:29  error  Trailing spaces not allowed  no-trailing-spaces
  661:30  error  Trailing spaces not allowed  no-trailing-spaces
```

madpink committed:
Removed trailing spaces

paglias committed:
correct deps loading and shrinkwrap

paglias committed:
chore(i18n): update locales

paglias committed:
chore(i18n) add overview and login incentives

paglias committed:
fix build by not requiring optional package in prod

Alys committed:
clarify two Notifications descriptions (onboarding, inactive account)

New words were supplied by Lemoness.

paglias committed:
client: add task component

paglias committed:
client: reorganize files, router and add inventory skeleton

mleah committed:
Adding "How to gift subscription" copy to Settings > Subscription Page:  Fixes Issue 8341 (#8386)

* Issue 8341 updating subscriber json and subscription settings page with gifting a subscription directions

* Issue-8341 updating the layout of the settings - subscription page

* Issue-8341 removing extra comma from gift subscription copy

* Issue-8341 removing extra spans in subscription jade file

declanramsay committed:
Add markdown formatting to task-edit title text (#8416)

SabreCat committed:
feat(content): new Gold quests (#8418)

and Wintery Skins, and fixes #8412

SabreCat committed:
chore(sprites): compile

SabreCat committed:
chore(i18n): update locales

SabreCat committed:
3.68.0

madpink committed:
updated tags API Doc

Finished updating the documentation

SabreCat committed:
fix(sprites): add large-1 to repo

TheHollidayInn committed:
Group plans misc fixes (#8388)

* Added notification for approval request in the group leaders language

* Added test for group task meta actions. Added sync when user claims

* Added tests for group task actions. Ensured assigned members are synce when added or removed

* Fixed approval required toggle

* Added support for users with comma in their name

* Fixed sync issue when user is approved and reloads the website

* Added advance options for group rewards

* Added back ticks to group claim message

* Fixed disappearing tasks that need approval

* Up chat limit to 400 for subbed groups

* Fixed line endings

* Updated activie subscription check

* Added group isSubscribed function

* Changed to isAfter

dumindux committed:
added message saying quest is started (#8107)

* added chat message saying quest has started

* added info to message

* added information as meta data

* fixed failing test case

* added new tests

abatula committed:
Critical Hits now affect boss damage fixes #5429 (#8092)

* Moved critical hit calculation from _addPoints() to _calculateDelta(). Added user as an input argument to _calculateDelta() so for critical hit calculation

* Changed test to expect task value of 1.5 after critical hit

* Revert "Moved critical hit calculation from _addPoints() to _calculateDelta(). Added user as an input argument to _calculateDelta() so for critical hit calculation"

This reverts commit 51b8ab6498d989d1f9494ac43c2605f3d4d14a35.

* Moved critical hit calculation to _changeTaskValue(). Use value stored in user._tmp.crit in _addPoints()

* Test is no longer affected by critical hits

* Removed unneeded comment

* Added WIP test of critical hits

* Want the crit function to return 2 to test critical hits

* Changed crit function to export as a function within an object so that it can be stubbed for testing. References to the crit() function were updated to call crit.crit() instead

* Added test for increased experience on critical hits

SabreCat committed:
fix(quest): correct pt 2 label

Alys committed:
remove Windows line breaks

paglias committed:
fix tests failures (#8422)

* fix tests failures

* make sure _meta is not public

* fix typo

* fix typo

gsacavdm committed:
Fix User > Profile showing {getProgressDisplay()} - fixes #8412 (#8421)

* Fix User > Profile showing {getProgressDisplay()}

* Remove bad nextRewardAt check

Mathwhiz committed:
apiDoc /chat Documentation (#8277)

* apiDoc /chat Documentation

* Add {NotFound} to FlagOwnMessage error

* Quick changes

* Update chat.js

madpink commented:
just checking in, is there anything else that I could/should do?

TheHollidayInn committed:
[WIP] Amazon refactor to lib (#8403)

* Moved amazon tests to folder

* Abstracted amazon payment code and added initial test

* Abstracted cancel and subscribe logic to amazon payment lib

* Added arg checks to checkout

* Added constants. Added more subscription test

* Added with arg checks to cancel

* Fixed linting issues

* Added integration tests for amazon subscribe cancel

* Added integration test for amazon checkout

* Added integration test for amazon subscribe

* Added coupon unit test

* Fixed lint

* Fixed minor test issue and changed header expectations

* Fixed line endings

eanplatter committed:
Add missing text to buttons on the party / guild invite modal - Fixes #8391 (#8402)

* Fix missing text in party / guild invite modal

* add tests and abstract logic

* use translation lib for test

SabreCat committed:
chore(news): misc Bailey

SabreCat committed:
chore(sprites): compile

SabreCat committed:
chore(i18n): update locales

SabreCat committed:
fix(typo): extra brackets

SabreCat committed:
3.69.0

SabreCat committed:
fix(docs): separate def blocks

SabreCat committed:
3.69.1

paglias commented:
@madpink can you rebase your branch to the current `develop` or merge it in? Because right now it's based on a `develop` commit that had some problems and the build is failing

Commit:
Merge remote-tracking branch 'HabitRPG/develop' into develop

madpink committed:
Merge pull request #1 from madpink/madpink-tags

Updated Tags APIDoc

Commit:
Update Tags API Doc

madpink commented:
Okay, so please tell me that I didn't destroy anything... the results were not quite what I was expecting, and my Tags changes also got added to this so I changed the title...

paglias commented:
@madpink the commit history is messed up but your commits didn't go lost. What I would suggest in your local install is:

```
git stash
git checkout develop
git fetch
git pull upstream develop
git checkout -b tasks-api-docs-changes
git cherry-pick 5359bc8ffc1f429cd5a44a92ee0323f8a02c0c32
git cherry-pick 4d7c451c910030826083c9305b5d2d8fd6f56e56
git cherry-pick 929ee3f562d19e2d4ae2f02e1d80995c75639918
git cherry-pick 130a83899c0cfdcb4919b3234a6b2fb951345571
git cherry-pick 5006f1f78ec702244a915942e4624765cbce1286
git cherry-pick bbd0d165d11f39ccbb6d12f18da8a633f400b8f8
git cherry pick 8d1a62bc7e3c317fb337f3a0abc023cf7551e463
```

which should reapply all tasks related commits in the `tasks-api-docs-changes` branch from which you can create another PR

Otherwise, if the above doesn't work, you should copy your changes somewhere, hard reset your develop branch and re-apply the changes

madpink commented:
I apologize, I really messed up my local repo and every try to fix it keeps throwing errors, even hard resetting... so I am removing my fork, cloning it again and adding my changes.
